dist/src/main/java/io/camunda/tasklist/TasklistModuleConfiguration.java

```diff
@@ -55,13 +61,47 @@ public class TasklistModuleConfiguration {
   // that the gateway is started first
   private final Gateway gateway;
 
+  @Autowired private SecurityConfiguration securityConfiguration;
+
   public TasklistModuleConfiguration(
       @Autowired(required = false) final Broker broker,
       @Autowired(required = false) final Gateway gateway) {
     this.broker = broker;
     this.gateway = gateway;
   }
 
+  /**
+   * Bean definition for `IdentityAuthorizationService` when the application is running under the
+   * "consolidated-auth" profile.
+   *
+   * <p>- The `@Profile("consolidated-auth")` annotation ensures that this bean is only loaded when
+   * the profile "consolidated-auth" is active.
+   *
+   * <p>This bean provides the **main authorization logic** when consolidated authentication is
+   * enabled.
+   */
+  @Bean
+  @Primary
+  @Profile("consolidated-auth")
+  public IdentityAuthorizationService consolidatedIdentityAuthorizationService() {
+    return new IdentityAuthorizationServiceImpl();
+  }
+
+  /**
+   * Fallback bean for `IdentityAuthorizationService`, used when the "consolidated-auth" profile is
+   * **not active**.
+   *
+   * <p>- This bean ensures that authorization is always available, even when consolidated
+   * authentication is not enabled. - It provides a **default, full-access authorization service**.
+   *
+   * <p>This bean declaration can be removed after the consolidated authentication is fully
+   * implemented
+   */
+  @Bean
```

Comment:
```suggestion
  @Bean
  @Profile("!consolidated-auth")
```

Reply:
Not so far, once I am using the @Primary bean in the other one.